internal/storage/caching/fast_stat_bucket.go

```diff
@@ -367,18 +367,22 @@ func (b *fastStatBucket) UpdateObject(
 func (b *fastStatBucket) DeleteObject(
 	ctx context.Context,
 	req *gcs.DeleteObjectRequest) (err error) {
-	b.invalidate(req.Name)
 	err = b.wrapped.DeleteObject(ctx, req)
+	if err != nil {
+		b.invalidate(req.Name)
```

Comment:
Why are we invalidating if entry is not deleted

I think it should be like this
```
If err != nil {
return
}
b.addNegativeEntry(req.Name)
```

Reply:
Thanks for your comment.

My thinking behind this was to not to assume that DeleteObject's failure means object is not deleted rather it might leave object's existence in precarious state from GCSFuse perspective for e.g. call to GCS is succeeded in the backend but somehow GCSFuse did not receive confirmation(momentary network loss) so in this scenario maintaining it in a cache is not a right thing to do as this makes an assumption that object is not deleted.

If we erase such entry from a cache then these assumptions regarding whether the object is deleted or not does not matter and next access to such objects will flow to GCS. 

Also, I do not think this should hamper performance as we are invalidating entry only in case of errors which I guess should be far and few. Still, I would like to hear from others' opinions about such a logic.

Reply:
Hmm okay.. then can you please write comment on it about the same.

Reply:
Added

Reply:
Still don't see the comment, seems you haven't pushed the changes?